### PR TITLE
Add integrity verification function to TrasferManager

### PR DESCRIPTION
**Pull request by Akifumi-Yanai**
Add the function of integrity verification of upload using MD5 to TransferManager.

In order to perform integrity verification with multipart upload, it is necessary to calculate MD5 for each part and attach `Content-MD5` header, but currently it can not be realized in TransferManager.
For that reason, I implemented TransferManager with MD5 calculation and `Content-MD5` header attached.

### CODE REVIEW

File: aws-cpp-sdk-transfer/include/aws/transfer/TransferManager.h [hunk 1]
```diff
@@ -60,6 +60,10 @@ namespace Aws
              * It is not a bug to use the same executor, but at least be aware that this is how the manager will be used.
              */
             Aws::Utils::Threading::Executor* transferExecutor;
+            /**
+             * If this is set to true, calculate content-md5 and set it to upload part request.
+             */
+            bool needsComputeContentMd5;
```
Comment (marcomagdy) [inline aws-cpp-sdk-transfer/include/aws/transfer/TransferManager.h]: Please, change the naming of the variable and the documentation to the following.

```cpp
/**
 * When true, TransferManager will calculate the MD5 digest of the content being uploaded.
 * The digest is sent to S3 via an HTTP header enabling the service to perform integrity checks.
 * This option is disabled by default.
 */
bool computeContentMD5;
```

**Comment by marcomagdy** [comment]
Actually, one more thing. Please change at least one of the integration tests to have this option turned on.

**Comment by Akifumi-Yanai** [comment]
Thanks for having a look at this. I applied your suggestion.

**Comment by marcomagdy** [comment]
@singku & @wps132230 can you take a look at this?

**Comment by nikt12** [comment]
Hello,
When it is planned to be merged?

**Comment by singku** [comment]
@nikt12 , Will resolve the conflict and run through our internal pipeline. Then get it merged.

**Comment by singku** [comment]
This is merged internally, will be out with next release.

**Pull request closed without merging**